platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,205 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[modularize_builds]]
+== Modularize Your Builds
+:keywords: structuring-builds
+
+Modularize your builds by creating subprojects.
+
+=== Explanation
+
+Splitting your build's source into multiple modules (Gradle projects) is essential for leveraging Gradle's automatic work avoidance and parallelization features.
+When a source file changes, Gradle only recompiles the affected projects.
+If all your sources reside in a single project, Gradle can't avoid recompilation and won't be able to run tasks in parallel.
+Splitting your build into multiple projects can provide additional performance benefits by minimizing each subproject's compilation classpath and ensuring annotation processors run only on the relevant source files.
+
+It makes sense to split your build's source into multiple projects _early_.
```

Comment:
"It makes sense" feels like an odd phrasing here.

Should we say, "Split your build's ..." or "Consider splitting your build's source ..."?

Reply:
Rephrasing to make this stronger and concise.